src/main/java/step03/controller/Main.java

```diff
@@ -0,0 +1,48 @@
+package step03.controller;
+
+import step03.domain.Ladder;
+import step03.domain.Matcher;
+import step03.domain.NamesOfResult;
+import step03.domain.Participants;
+import step03.view.InputView;
+import step03.view.ResultView;
+import strategy.RandomPointStrategy;
+
+public class Main {
+    public static void main(String[] args) {
+
+        Participants participants = readParticipants();
+        NamesOfResult namesOfResult = readNamesOfResults();
+
+        Matcher.Builder matcherBuilder = Matcher.Builder.of(participants)
+                .namesOfResult(namesOfResult);
+
+        Integer countOfSteps = InputView.readCountOfSteps();
+
+        Ladder ladder = Ladder.of(countOfSteps, participants.size(), RandomPointStrategy.of());
+        ResultView.printLadderWithNames(participants, namesOfResult, ladder.getLadder());
+
+        Matcher matcher = matcherBuilder.matchedTable(ladder.getResultTable()).build();
+
+        printResult(matcher);
+    }
+
+    public static Participants readParticipants() {
+        String namesOfParticipants = InputView.readNamesOfParticipants();
+        return Participants.of(namesOfParticipants);
+    }
+
+    public static NamesOfResult readNamesOfResults() {
+        String resultsOfLADDER = InputView.readResultsOfLadder();
+        return NamesOfResult.of(resultsOfLADDER);
+    }
+
+    public static void printResult(Matcher matcher) {
+        String target = InputView.readNamesOfParticipant();
+        while(!target.equals("all")) {
```

Comment:
저는 사용자의 입력값을 절대로 믿지 않는 비관적인(?) 입장이기에 입력값인 target이 언제든지 null로 들어올 수 있다는 의심을 항상 하고있습니다 ㅎㅎ
그렇기에 target.equals()를 하게 되면 null 참조로 인해 NPE 가 발생할 수도 있다고 생각해요!
다음과 같이 변경하면 좋을 것 같아요 :)

```suggestion
        String target = InputView.readNamesOfParticipant();
        while (!"all".equals(target)) {
```

Reply:
8c686f7
그런 상황이 있을수도 있군요! 꿀팁 감사합니다